src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -745,6 +745,34 @@ public async Task GetAsync_InvalidHeaderNameValue_ThrowsHttpRequestException(str
             }, server => server.AcceptConnectionSendCustomResponseAndCloseAsync($"HTTP/1.1 200 OK\r\n{invalidHeader}\r\nContent-Length: 11\r\n\r\nhello world"));
         }
 
+        [Theory]
+        [InlineData(false, false)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(true, true)]
+        public async Task GetAsync_IncompleteData_ThrowsHttpRequestException(bool failDuringHeaders, bool getString)
+        {
+            if (IsWinHttpHandler)
+            {
+                // [ActiveIssue(39136)]
+                return;
+            }
+
+            await LoopbackServer.CreateClientAndServerAsync(async uri =>
```

Comment:
Would be nice to make this work for HTTP2 as well. I'm fine if you want to defer that, though.

Reply:
Ah, right, will fix tomorrow.

Reply:
Actually, doesn't look like we have anything on the GenericLoopbackServer today that enables this, at least for failDuringHeaders=true.  We can follow-up on this separately, along with the vast majority of other tests in this file that are 1.1-specific.  The good news is this logic is entirely contained in HttpClient itself, which is agnostic as to HTTP version.

Reply:
Think of this as an opportunity to improve GenericLoopbackServer :)

(seriously though, i'm fine if these are 1.1 specific for now)